### ISSUE TITLE
fakeroot option does not work with hidepid procfs

### ISSUE THREAD

**Issue by dc-fukuoka** [apptainer/singularity#4633]
### Version of Singularity:

What version of Singularity are you using? Run:

```
$ singularity version
3.4.1
```

### Expected behavior

What did you expect to see when you do...?

```
$ singularity shell -f -w ubuntu/
Singularity ubuntu:~>
```

### Actual behavior

```
singularity shell -f -w ubuntu/
ERROR  : Failed to change directory to /proc/21020: No such file or directory
```

What actually happend? Why was it incorrect?

This error occurs when hidepid=1 or 2 option is used in the procfs with fakeroot option.
Is it possible to use fakeroot with hidpid?

### Steps to reproduce this behavior

How can others reproduce this issue/problem?

Mount procfs with hidepid=1 or 2 and execute `singularity -f`

### What OS/distro are you running

```
$ cat /etc/os-release
NAME="SLES"
VERSION="12-SP2"
VERSION_ID="12.2"
PRETTY_NAME="SUSE Linux Enterprise Server 12 SP2"
ID="sles"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:12:sp2"

```

### How did you install Singularity

Write here how you installed Singularity. Eg. RPM, source.

From the source

**Comment by cclerget** [comment]
Hi @dc-fukuoka , worked on fix #4634, If you can give a try and let me know ?

In general `hidepid` prevented Singularity from working correctly, after some tests there is one case which couldn't work with `hidepid` option: joining/stopping an instance started with setuid workflow, it works fine for root user, `--fakeroot` or `--userns` options but not for user with setuid workflow.

**Comment by dc-fukuoka** [comment]
Hi @cclerget , thank you for creating the patch.
I confirmed that it works fine!
